### ISSUE TITLE
'import/no-unresolved' always additionally causes false positive 'import/extensions'

### ISSUE THREAD

**Issue by mbonaci** [benmosher/eslint-plugin-import#2153]
This is more of a low-priority FYI, given that `no-unresolved` always gets fixed right away, so the issue is inconsequential.

![image](https://user-images.githubusercontent.com/497926/125773310-c49a2956-2cc4-493e-99ec-7bed552393ad.png)

I haven't specified `import/extensions` rule and thus the default `"never"` should be in effect.  
That's how it behaves, **except** when there's an actual (legitimate) issue with import resolution.

From `.eslintrc`:
```
"parser": "@babel/eslint-parser",
"extends": "airbnb",
...
"import/no-named-as-default": 1,
"import/order": "off",
"import/prefer-default-export": "off",
```

From `devDependencies`:
```
"@babel/eslint-parser": "^7.14.7",
"eslint": "^7.30.0",
"eslint-config-airbnb": "^18.2.1",
"eslint-import-resolver-webpack": "^0.13.1",
"eslint-loader": "^4.0.2",
"eslint-plugin-import": "^2.23.4",
"eslint-plugin-react": "^7.24.0",
```

**Comment by soryy708** [comment]
It's unclear to me what exactly is the problem, how do I try to reproduce this on latest plugin and eslint version?
